src/modules/project/pages/projectView/views/projectMainBody/utils.tsx

```diff
@@ -0,0 +1,14 @@
+export const generateTwitterShareUrl = (tweetTemplate: string, projectLink: string): string => {
+  const tweet = tweetTemplate.replace('{{projectLink}}', projectLink)
+  return `https://twitter.com/intent/tweet?text=${encodeURIComponent(tweet)}`
+}
+
+export const shareOnTwitter = (tweetTemplates: string[], projectName: string, t: any) => {
+  const projectLink = `${window.location.origin}/project/${projectName}`
+
+  const randomKey = tweetTemplates[Math.floor(Math.random() * tweetTemplates.length)] || ''
+  const tweetTemplate = t(randomKey)
+  const twitterUrl = generateTwitterShareUrl(tweetTemplate, projectLink)
+
+  window.open(twitterUrl, '_blank')
+}
```

Comment:
last bit, the window.open doesnot work on certain privacy focused mobile browsers. it does not open it considering it as a popup, as it was triggered from JS instead by a user through a button click.

I suggest, move this code back to the button as an <a> tag, that opens a new window.

Reply:
on which browser did you test this so I can test it myself if it works?